### ISSUE TITLE
Convert `padding_mode` into Enum

### ISSUE THREAD

**Issue by vfdev-5** [pytorch/vision#6250]
Currently `padding_mode` is a string: "constant", "edge", "reflect", "symmetric" and in the codebase we already have enums like InterpolationMode and others in prototype space.

Let's create an enum for padding mode and convert the type into it while keeping BC and going a deprecation cycles etc.

**Comment by vadimkantorov** [comment]
This would also introduce inconsistency with core PyTorch F.pad, F.conv2d padding argument (which has string values of same/valid)

**Comment by vfdev-5** [comment]
@vadimkantorov there is already such difference for interpolation mode

**Comment by vadimkantorov** [comment]
yes, and the question stays the same: is introducing more inconsistencies the way forward? is pytorch core going to adopt enums? what're core teams thoughts on this? why does core pytorch not adopt enums? and if not, what are their arguments and are they valid for torchvision.

i mean, it's just interesting to know what was the outcome of these internal discussions

**Comment by NicolasHug** [comment]
I tend to agree that consistency with core is valuable here.

Using enums has the added inconvenience that it requires an extra import from users. @vfdev-5 what would be the key reason for us to start adopting enums for `padding_mode` here?
